### PR TITLE
scripts: west_commands: ncs-provision locked key

### PR DESCRIPTION
**Pull request by michalek-no**
adds ability to upload key as locked.

ref.: NCSDK-30867

### CODE REVIEW

**Review by nvlsianpu — APPROVED** [review]

File: scripts/west_commands/ncs-provision.py [hunk 1]
```diff
@@ -34,6 +36,8 @@ def do_add_parser(self, parser_adder):
             "-k", "--key", type=Path, action='append', dest="keys",
             help="Input .pem file with ED25519 private key"
         )
+        upload_parser.add_argument("-p", "--policy", type=str, help="Keys policy",
+                                   choices=["default", "lock"], default="default")
```
Comment (fundakol) [inline scripts/west_commands/ncs-provision.py:40]: It's not clear what `default` means, why don't you use `choices=["revoked", "locked"]` and
```python
  "-r",
  args.policy.uppder(),
  "-v",
```

Comment (michalek-no) [reply scripts/west_commands/ncs-provision.py:40]: translation table on top

Comment (gchwier) [reply scripts/west_commands/ncs-provision.py:40]: agree with @fundakol , don't need extra table, and why to change parameter names from `nrfprovision` tool?
you can use:
```
        upload_parser.add_argument("-p", "--policy", type=str.upper, help="Keys policy",
            choices=["REVOKED", "ROTATING", "LOCKED"], default="REVOKED")
```

Comment (michalek-no) [reply scripts/west_commands/ncs-provision.py:40]: Name 'revoked' suggests it's dead right away. It should be named differently earlier on but here we are.

**Review by gchwier — APPROVED** [review]

**Review by fundakol — APPROVED** [review]

**Comment by michalek-no** [comment]
rebase

**Comment by shanthanordic** [comment]
@nvlsianpu  should this be backported to 2.8 branch as well ?

**Comment by nvlsianpu** [comment]
@shanthanordic Yes it should - it's mitigation to known issue, backward compatible.

**Pull request merged**